### PR TITLE
Add some null checks in DatasetTrackerChronicle

### PR DESCRIPTION
**Pull request by WeatherGod**

### CODE REVIEW

**Comment by WeatherGod** [comment]
Currently trying this out in my k8s deployment since it is able to trigger failures to create a dataset tracker.

**Comment by WeatherGod** [comment]
Ok, it reduces the amount of redundant stack traces in the logs, and it also helps the services to properly (or at least more correctly) error out rather than hang or otherwise barf.

File: tdcommon/src/main/java/thredds/server/catalog/tracker/DatasetTrackerChronicle.java [hunk 1]
```diff
@@ -207,19 +210,29 @@ else if (!path.equals(accessPath)) { // LOOK must put all for restrict
 
     // changed = true;
     DatasetExt dsext = new DatasetExt(catId, dataset.getRestrictAccess(), ncml);
+    if (datasetMap == null) {
+      catalogInitLog.error("Catalog chronicle map not properly initialized. Can't track dataset {}", dataset);
```
Comment (WeatherGod) [inline tdcommon/src/main/java/thredds/server/catalog/tracker/DatasetTrackerChronicle.java:214]: I'll note that this doesn't print out the name of the dataset, it prints out a repr of the dataset instance. I just followed the same pattern as above, if that's good enough.

**Comment by WeatherGod** [comment]
I wouldn't know how to make tests for this since I don't know how to fail the creation of the chronicle file. Maybe point it to a directory that doesn't exist? Don't know if it makes directories automatically.

**Comment by WeatherGod** [comment]
For the record from my logs:
```
# cat content/thredds/logs/catalogInit.log 
You are currently running TDS version 5.5-SNAPSHOT - 2024-03-28T04:58:45+0000
Latest Available TDS Version Info:
    latest development version = 5.5-SNAPSHOT
    latest release version = 5.4

=========================================================================================
ConfigCatalogInitialization readMode=always isStartup=true
DatasetTrackerChronicle failed on '/usr/local/tomcat/content/thredds/cache/catalog/chronicle.datasets.dat.1', delete catalog cache and reload 
java.io.IOException: Not enough space to fallocate 8 MiB to / unallocated was 0 MiB
	at net.openhft.chronicle.hash.impl.VanillaChronicleHash.fallocate(VanillaChronicleHash.java:1123) ~[chronicle-map-3.23.5.jar:3.23.5]
	at net.openhft.chronicle.hash.impl.VanillaChronicleHash.map(VanillaChronicleHash.java:1103) ~[chronicle-map-3.23.5.jar:3.23.5]
	at net.openhft.chronicle.hash.impl.VanillaChronicleHash.createMappedStoreAndSegments(VanillaChronicleHash.java:515) ~[chronicle-map-3.23.5.jar:3.23.5]
	at net.openhft.chronicle.map.ChronicleMapBuilder.createWithNewFile(ChronicleMapBuilder.java:1843) ~[chronicle-map-3.23.5.jar:3.23.5]
	at net.openhft.chronicle.map.ChronicleMapBuilder.createWithFile(ChronicleMapBuilder.java:1741) ~[chronicle-map-3.23.5.jar:3.23.5]
	at net.openhft.chronicle.map.ChronicleMapBuilder.createPersistedTo(ChronicleMapBuilder.java:1586) ~[chronicle-map-3.23.5.jar:3.23.5]
	at thredds.server.catalog.tracker.DatasetTrackerChronicle.open(DatasetTrackerChronicle.java:138) ~[classes/:5.5-SNAPSHOT]
	at thredds.server.catalog.tracker.DatasetTrackerChronicle.<init>(DatasetTrackerChronicle.java:79) [classes/:5.5-SNAPSHOT]
	at thredds.server.catalog.tracker.DatasetTrackerChronicle.<init>(DatasetTrackerChronicle.java:68) [classes/:5.5-SNAPSHOT]
	at thredds.core.ConfigCatalogInitialization.reread(ConfigCatalogInitialization.java:149) [classes/:5.5-SNAPSHOT]
	at thredds.core.ConfigCatalogInitialization.init(ConfigCatalogInitialization.java:131) [classes/:5.5-SNAPSHOT]
	at thredds.server.config.TdsInit.onApplicationEvent(TdsInit.java:149) [classes/:5.5-SNAPSHOT]
	at thredds.server.config.TdsInit.onApplicationEvent(TdsInit.java:67) [classes/:5.5-SNAPSHOT]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:178) [spring-context-5.3.32.jar:5.3.32]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:171) [spring-context-5.3.32.jar:5.3.32]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:145) [spring-context-5.3.32.jar:5.3.32]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:429) [spring-context-5.3.32.jar:5.3.32]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:386) [spring-context-5.3.32.jar:5.3.32]
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:949) [spring-context-5.3.32.jar:5.3.32]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:594) [spring-context-5.3.32.jar:5.3.32]
	at org.springframework.web.context.ContextLoader.configureAndRefreshWebApplicationContext(ContextLoader.java:399) [spring-web-5.3.32.jar:5.3.32]
	at org.springframework.web.context.ContextLoader.initWebApplicationContext(ContextLoader.java:278) [spring-web-5.3.32.jar:5.3.32]
	at org.springframework.web.context.ContextLoaderListener.contextInitialized(ContextLoaderListener.java:103) [spring-web-5.3.32.jar:5.3.32]
	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4441) [catalina.jar:8.5.100]
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:4899) [catalina.jar:8.5.100]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:171) [catalina.jar:8.5.100]
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:710) [catalina.jar:8.5.100]
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:687) [catalina.jar:8.5.100]
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:660) [catalina.jar:8.5.100]
	at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1176) [catalina.jar:8.5.100]
	at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1889) [catalina.jar:8.5.100]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
DatasetTrackerChronicle failed on '/usr/local/tomcat/content/thredds/cache/catalog/chronicle.datasets.dat.1', delete catalog cache and reload 
java.io.IOException: Not enough space to fallocate 8 MiB to / unallocated was 0 MiB
	at net.openhft.chronicle.hash.impl.VanillaChronicleHash.fallocate(VanillaChronicleHash.java:1123) ~[chronicle-map-3.23.5.jar:3.23.5]
	at net.openhft.chronicle.hash.impl.VanillaChronicleHash.map(VanillaChronicleHash.java:1103) ~[chronicle-map-3.23.5.jar:3.23.5]
	at net.openhft.chronicle.hash.impl.VanillaChronicleHash.createMappedStoreAndSegments(VanillaChronicleHash.java:515) ~[chronicle-map-3.23.5.jar:3.23.5]
	at net.openhft.chronicle.map.ChronicleMapBuilder.createWithNewFile(ChronicleMapBuilder.java:1843) ~[chronicle-map-3.23.5.jar:3.23.5]
	at net.openhft.chronicle.map.ChronicleMapBuilder.createWithFile(ChronicleMapBuilder.java:1741) ~[chronicle-map-3.23.5.jar:3.23.5]
	at net.openhft.chronicle.map.ChronicleMapBuilder.createPersistedTo(ChronicleMapBuilder.java:1586) ~[chronicle-map-3.23.5.jar:3.23.5]
	at thredds.server.catalog.tracker.DatasetTrackerChronicle.open(DatasetTrackerChronicle.java:138) ~[classes/:5.5-SNAPSHOT]
	at thredds.server.catalog.tracker.DatasetTrackerChronicle.reinit(DatasetTrackerChronicle.java:120) [classes/:5.5-SNAPSHOT]
	at thredds.server.catalog.tracker.DatasetTrackerChronicle.<init>(DatasetTrackerChronicle.java:85) [classes/:5.5-SNAPSHOT]
	at thredds.server.catalog.tracker.DatasetTrackerChronicle.<init>(DatasetTrackerChronicle.java:68) [classes/:5.5-SNAPSHOT]
	at thredds.core.ConfigCatalogInitialization.reread(ConfigCatalogInitialization.java:149) [classes/:5.5-SNAPSHOT]
	at thredds.core.ConfigCatalogInitialization.init(ConfigCatalogInitialization.java:131) [classes/:5.5-SNAPSHOT]
	at thredds.server.config.TdsInit.onApplicationEvent(TdsInit.java:149) [classes/:5.5-SNAPSHOT]
	at thredds.server.config.TdsInit.onApplicationEvent(TdsInit.java:67) [classes/:5.5-SNAPSHOT]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:178) [spring-context-5.3.32.jar:5.3.32]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:171) [spring-context-5.3.32.jar:5.3.32]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:145) [spring-context-5.3.32.jar:5.3.32]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:429) [spring-context-5.3.32.jar:5.3.32]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:386) [spring-context-5.3.32.jar:5.3.32]
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:949) [spring-context-5.3.32.jar:5.3.32]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:594) [spring-context-5.3.32.jar:5.3.32]
	at org.springframework.web.context.ContextLoader.configureAndRefreshWebApplicationContext(ContextLoader.java:399) [spring-web-5.3.32.jar:5.3.32]
	at org.springframework.web.context.ContextLoader.initWebApplicationContext(ContextLoader.java:278) [spring-web-5.3.32.jar:5.3.32]
	at org.springframework.web.context.ContextLoaderListener.contextInitialized(ContextLoaderListener.java:103) [spring-web-5.3.32.jar:5.3.32]
	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4441) [catalina.jar:8.5.100]
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:4899) [catalina.jar:8.5.100]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:171) [catalina.jar:8.5.100]
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:710) [catalina.jar:8.5.100]
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:687) [catalina.jar:8.5.100]
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:660) [catalina.jar:8.5.100]
	at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1176) [catalina.jar:8.5.100]
	at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1889) [catalina.jar:8.5.100]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
ConfigCatalogInitializion datasetTracker database does not exist, set readMode to=always
ConfigCatalogInit: initializing 1 root catalogs.
Checking catalogRoot = catalog.xml
-------readCatalog(): path=catalog.xml
Added global service Service{ name=latest, base=, typeS=Resolver}
Added global service Service{ name=all, base=, typeS=compound
  Service{ name=odap, base=/thredds/dodsC/, typeS=OpenDAP,desc=Access dataset through OPeNDAP using the DAP2 protocol.}
  Service{ name=wcs, base=/thredds/wcs/, typeS=WCS,desc=Supports access to geospatial data as 'coverages'.}
  Service{ name=wms, base=/thredds/wms/, typeS=WMS,desc=Supports access to georegistered map images from geoscience datasets.}
  Service{ name=ncss, base=/thredds/ncss/, typeS=NetcdfSubset,desc=A web service for subsetting CDM scientific datasets.}
  Service{ name=ncml, base=/thredds/ncml/, typeS=NCML,desc=Provide NCML representation of a dataset.}
}

Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@3018399a
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@69a982c4
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@5a113867
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@3f68e5f0
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@4e4ddf6e
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@6b57dbc1
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@38d604cc
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@40e92143
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@60ff309f
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@35a37692
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@3ddbe48e
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@3431294f
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@2d538c9e
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@36a6840
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@61f71337
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@5cb12517
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@6db77563
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@3d4f28fd
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@6478eac8
Catalog chronicle map not properly initialized. Can't track dataset thredds.client.catalog.Dataset@31b4e2cd

ConfigCatalogInitializion stats
ConfigCatalogInitialization started 2024-03-28T21:33:11.115Z took 0.177000 secs using readMode=always
         catalogs=1
         datasets=20
  trackedDatasets=20
           restrict=0
            hasNcml=20

      dataRoot=0
    featCollect=0
    datasetScan=0
    catalogScan=0
    datasetRoot=0

DatasetExt.total_count 0
DatasetExt.total_nbytes 0
DatasetExt.avg_nbytes     0

restrict (0)

nAccess (1)
             5: count = 20

serviceType (5)
    Service{ name=ncml, base=/thredds/ncml/, typeS=NCML,desc=Provide NCML representation of a dataset.}: count = 20
    Service{ name=ncss, base=/thredds/ncss/, typeS=NetcdfSubset,desc=A web service for subsetting CDM scientific datasets.}: count = 20
    Service{ name=odap, base=/thredds/dodsC/, typeS=OpenDAP,desc=Access dataset through OPeNDAP using the DAP2 protocol.}: count = 20
    Service{ name=wcs, base=/thredds/wcs/, typeS=WCS,desc=Supports access to geospatial data as 'coverages'.}: count = 20
    Service{ name=wms, base=/thredds/wms/, typeS=WMS,desc=Supports access to georegistered map images from geoscience datasets.}: count = 20

ncmlAggSize (1)
             0: count = 20

ConfigCatalogInitializion finished took=303 msecs
```

And me making a few attempts at accessing metadata:
```
2024-03-28T21:39:32.783 +0000 [    397484][      60] INFO  - threddsServlet - Remote host: 10.16.1.62 - Request: "GET /thredds/catalog/catalog.html?dataset=era5-monthly-anomaly HTTP/1.1"
2024-03-28T21:39:32.919 +0000 [    397620][      60] INFO  - threddsServlet - Request Completed - 200 - -1 - 136
2024-03-28T21:39:35.398 +0000 [    400099][      61] INFO  - threddsServlet - Remote host: 10.16.1.62 - Request: "GET /thredds/wms/era5/monthly-anomaly.nc?service=WMS&version=1.3.0&request=GetCapabilities HTTP/1.1"
2024-03-28T21:39:35.398 +0000 [    400099][      61] INFO  - thredds.server.wms.ThreddsWmsServlet - Did not find era5/monthly-anomaly.nc in the cache
2024-03-28T21:39:35.570 +0000 [    400271][      62] INFO  - threddsServlet - Remote host: 10.16.12.235 - Request: "GET /thredds/catalog/catalog.html HTTP/1.1"
2024-03-28T21:39:35.582 +0000 [    400283][      62] INFO  - threddsServlet - Request Completed - 200 - -1 - 12
2024-03-28T21:39:45.570 +0000 [    410271][      63] INFO  - threddsServlet - Remote host: 10.16.12.235 - Request: "GET /thredds/catalog/catalog.html HTTP/1.1"
2024-03-28T21:39:45.584 +0000 [    410285][      63] INFO  - threddsServlet - Request Completed - 200 - -1 - 14
2024-03-28T21:39:55.570 +0000 [    420271][      64] INFO  - threddsServlet - Remote host: 10.16.12.235 - Request: "GET /thredds/catalog/catalog.html HTTP/1.1"
2024-03-28T21:39:55.583 +0000 [    420284][      64] INFO  - threddsServlet - Request Completed - 200 - -1 - 13
2024-03-28T21:40:05.569 +0000 [    430270][      65] INFO  - threddsServlet - Remote host: 10.16.12.235 - Request: "GET /thredds/catalog/catalog.html HTTP/1.1"
2024-03-28T21:40:05.580 +0000 [    430281][      65] INFO  - threddsServlet - Request Completed - 200 - -1 - 11
2024-03-28T21:40:15.570 +0000 [    440271][      66] INFO  - threddsServlet - Remote host: 10.16.12.235 - Request: "GET /thredds/catalog/catalog.html HTTP/1.1"
2024-03-28T21:40:15.583 +0000 [    440284][      66] INFO  - threddsServlet - Request Completed - 200 - -1 - 14
2024-03-28T21:40:25.569 +0000 [    450270][      67] INFO  - threddsServlet - Remote host: 10.16.12.235 - Request: "GET /thredds/catalog/catalog.html HTTP/1.1"
2024-03-28T21:40:25.581 +0000 [    450282][      67] INFO  - threddsServlet - Request Completed - 200 - -1 - 12
2024-03-28T21:40:27.740 +0000 [    452441][      68] INFO  - threddsServlet - Remote host: 10.16.1.62 - Request: "GET /thredds/ncml/era5/monthly-anomaly.nc HTTP/1.1"
2024-03-28T21:40:27.766 +0000 [    452467][      68] ERROR - thredds.server.metadata.controller.NcmlController - Error in NCML: null
java.lang.IllegalStateException: Cannot call sendError() after the response has been committed
	at org.apache.catalina.connector.ResponseFacade.checkCommitted(ResponseFacade.java:495) ~[catalina.jar:8.5.100]
	at org.apache.catalina.connector.ResponseFacade.sendError(ResponseFacade.java:342) ~[catalina.jar:8.5.100]
	at javax.servlet.http.HttpServletResponseWrapper.sendError(HttpServletResponseWrapper.java:111) ~[servlet-api.jar:?]
	at javax.servlet.http.HttpServletResponseWrapper.sendError(HttpServletResponseWrapper.java:111) ~[servlet-api.jar:?]
	at thredds.servlet.filter.TdsServletResponseWrapper.sendError(TdsServletResponseWrapper.java:48) ~[classes/:5.5-SNAPSHOT]
	at thredds.server.metadata.controller.NcmlController.handleMetadataRequest(NcmlController.java:109) [tds-plugin-jar-with-dependencies.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205) [spring-web-5.3.32.jar:5.3.32]
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:150) [spring-web-5.3.32.jar:5.3.32]
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117) [spring-webmvc-5.3.32.jar:5.3.32]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:898) [spring-webmvc-5.3.32.jar:5.3.32]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:809) [spring-webmvc-5.3.32.jar:5.3.32]
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87) [spring-webmvc-5.3.32.jar:5.3.32]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1072) [spring-webmvc-5.3.32.jar:5.3.32]
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:965) [spring-webmvc-5.3.32.jar:5.3.32]
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006) [spring-webmvc-5.3.32.jar:5.3.32]
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:898) [spring-webmvc-5.3.32.jar:5.3.32]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:489) [servlet-api.jar:?]
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883) [spring-webmvc-5.3.32.jar:5.3.32]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:583) [servlet-api.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:212) [catalina.jar:8.5.100]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:156) [catalina.jar:8.5.100]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51) [tomcat-websocket.jar:8.5.100]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:181) [catalina.jar:8.5.100]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:156) [catalina.jar:8.5.100]
	at org.apache.catalina.filters.CorsFilter.handleNonCORS(CorsFilter.java:333) [catalina.jar:8.5.100]
	at org.apache.catalina.filters.CorsFilter.doFilter(CorsFilter.java:155) [catalina.jar:8.5.100]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:181) [catalina.jar:8.5.100]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:156) [catalina.jar:8.5.100]
	at org.apache.catalina.filters.HttpHeaderSecurityFilter.doFilter(HttpHeaderSecurityFilter.java:129) [catalina.jar:8.5.100]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:181) [catalina.jar:8.5.100]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:156) [catalina.jar:8.5.100]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:337) [spring-security-web-5.7.12.jar:5.7.12]
	at thredds.servlet.filter.RequestBracketingLogMessageFilter.doFilter(RequestBracketingLogMessageFilter.java:50) [classes/:5.5-SNAPSHOT]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346) [spring-security-web-5.7.12.jar:5.7.12]
	at thredds.servlet.filter.RequestQueryFilter.doFilter(RequestQueryFilter.java:90) [classes/:5.5-SNAPSHOT]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346) [spring-security-web-5.7.12.jar:5.7.12]
	at thredds.servlet.filter.HttpHeadFilter.doFilter(HttpHeadFilter.java:47) [classes/:5.5-SNAPSHOT]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346) [spring-security-web-5.7.12.jar:5.7.12]
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:221) [spring-security-web-5.7.12.jar:5.7.12]
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:186) [spring-security-web-5.7.12.jar:5.7.12]
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:354) [spring-web-5.3.32.jar:5.3.32]
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:267) [spring-web-5.3.32.jar:5.3.32]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:181) [catalina.jar:8.5.100]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:156) [catalina.jar:8.5.100]
	at org.apache.logging.log4j.web.Log4jServletFilter.doFilter(Log4jServletFilter.java:71) [log4j-web-2.17.1.jar:2.17.1]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:181) [catalina.jar:8.5.100]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:156) [catalina.jar:8.5.100]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:168) [catalina.jar:8.5.100]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90) [catalina.jar:8.5.100]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483) [catalina.jar:8.5.100]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130) [catalina.jar:8.5.100]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93) [catalina.jar:8.5.100]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:679) [catalina.jar:8.5.100]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) [catalina.jar:8.5.100]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:346) [catalina.jar:8.5.100]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:617) [tomcat-coyote.jar:8.5.100]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63) [tomcat-coyote.jar:8.5.100]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:934) [tomcat-coyote.jar:8.5.100]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1690) [tomcat-coyote.jar:8.5.100]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52) [tomcat-coyote.jar:8.5.100]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) [tomcat-util.jar:8.5.100]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) [tomcat-util.jar:8.5.100]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63) [tomcat-util.jar:8.5.100]
	at java.lang.Thread.run(Thread.java:829) [?:?]
2024-03-28T21:40:27.768 +0000 [    452469][      68] INFO  - threddsServlet - Request Completed - 500 - -1 - 28
2024-03-28T21:40:32.909 +0000 [    457610][      69] INFO  - threddsServlet - Remote host: 10.16.1.62 - Request: "GET /thredds/dodsC/era5/monthly-anomaly.nc.html HTTP/1.1"
2024-03-28T21:40:32.920 +0000 [    457621][      69] INFO  - threddsServlet - Request Completed - 404 - -1 - 11
```

**Comment by WeatherGod** [comment]
And, to be clear, this doesn't fix #484, it just makes thredds fail a bit more gracefully and produces slightly cleaner logs in this apparently rare situation.

File: tdcommon/src/main/java/thredds/server/catalog/tracker/DatasetTrackerChronicle.java [hunk 2]
```diff
@@ -132,6 +132,9 @@ private void open() throws IOException {
     ChronicleMapBuilder<String, DatasetExt> builder = ChronicleMapBuilder.of(String.class, DatasetExt.class)
         .averageValueSize(averageValueSize).entries(maxDatasets).averageKeySize(averagePathLength)
         .valueMarshaller(DatasetExtBytesMarshaller.INSTANCE).skipCloseOnExitHook(true);
+    if (builder == null) {
+      throw new IOException("Failed to create Dataset catalog tracker");
+    }
```
Comment (WeatherGod) [inline tdcommon/src/main/java/thredds/server/catalog/tracker/DatasetTrackerChronicle.java:137]: This check might be superfluous. I'm not sure if ChronicleMapBuilder.of() can even return a null. At the moment for my situation, it throws an exception instead.

**Review by tdrwenski — CHANGES_REQUESTED** [review]

**Review by tdrwenski — APPROVED** [review]

**Pull request merged**